<launch>  
    <!-- Arguments -->
    <arg name="host" default="localhost" />
    <arg name="tcp_role" default="host" />
    <arg name="port" default="3000" />
    <arg name="cert_path" />
    <arg name="key_path" />
    <arg name="cert_verify_path" />

    <node pkg="rostopic" type="rostopic" name="float_host" args="pub -r 500 /float_host std_msgs/Float32 1.0" output="screen"/>
    <node pkg="rostopic" type="rostopic" name="string_host" args="pub -r 500 /string_host std_msgs/String string" output="screen"/>

    <!-- Sockets -->
    <node name="tcp_host" pkg="tcp_socket" type="socket_bidirectional_node.py" output="screen">
        <param name="tcp_role" type="string" value="$(arg tcp_role)" />
        <param name="host" type="string" value="$(arg host)" />
        <param name="port" type="int" value="$(arg port)" />
        <param name="topic_yaml" type="string" value="$(find tcp_socket)/config/socket_topics.yaml" />
        <param name="cert_path" type="string" value="$(arg cert_path)" />
        <param name="key_path" type="string" value="$(arg key_path)" />
        <param name="cert_verify_path" type="string" value="$(arg cert_verify_path)" />
    </node>

</launch>